Controller/ProfilerController.php

```diff
@@ -11,27 +12,29 @@
 use PDO;
 use PDOStatement;
 use Symfony\Bridge\Doctrine\DataCollector\DoctrineDataCollector;
-use Symfony\Component\DependencyInjection\ContainerAwareInterface;
-use Symfony\Component\DependencyInjection\ContainerInterface;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpKernel\Profiler\Profiler;
 use Symfony\Component\VarDumper\Cloner\Data;
 use Throwable;
+use Twig\Environment;
 
 use function assert;
 use function stripos;
 
-class ProfilerController implements ContainerAwareInterface
+class ProfilerController
 {
-    /** @var ContainerInterface */
-    private $container;
-
-    /**
-     * {@inheritDoc}
-     */
-    public function setContainer(?ContainerInterface $container = null)
+    /** @var Environment */
+    private $twig;
+    /** @var Registry */
+    private $registry;
+    /** @var Profiler */
+    private $profiler;
+
+    public function __construct(Environment $twig, Registry $registry, Profiler $profiler)
```

Comment:
whats the BC policy for this class? :thinking: Maybe not likely that someone extended it? But possible...

Reply:
I would tag it as `@internal` (and document in the changelog that we removed that class from the public API because it was never meant to be part of it)

Reply:
Our BC policy is whatever we feel is right. We try to avoid this kind of things in patch releases, but if given issue is big enough like now, we can do this. We are purposefully vague to not limit ourselves. We don't have same policy as Symfony. Gymnastics of making this BC compatible is not worth it here for very unlikely scenario of someone extending this class.

Reply:
ok then let's tag it as `@internal`? To make it more obvious that people should not extend this

Reply:
done